awx/main/tests/functional/test_inventory_source_injectors.py

```diff
@@ -289,7 +291,8 @@ def substitute_run(envvars=None, **_kw):
             for f_name in expected_file_list:
                 with open(os.path.join(files_dir, f_name), 'r') as f:
                     ref_content = f.read()
-                    assert ref_content == content[f_name], f_name
+                    assert yaml.load(ref_content, Loader=SafeLoader) == \
+                        yaml.load(content[f_name], Loader=SafeLoader), f_name
```

Comment:
This isn't the right place to address this in my book. The content generated for the inventory configs is outwardly-facing server behavior, this is just testing it. You should be more concerned that the generated configs do not pass YAML lint than that the test files do not pass.

To fix that, the `safe_dump` command here should be amended so that it always produces an output that passes YAML lint.

https://github.com/ansible/awx/blob/8116ec8e1f7d294fbba326397644453da16d3cbb/awx/main/models/inventory.py#L1895-L1899

Then, I don't see any reason that this assertion should change, but the modifications to the test data files should remain and and the tests should pass because then everything will conform to the standard.

Reply:
> This isn't the right place to address this in my book. The content generated for the inventory configs is outwardly-facing server behavior, this is just testing it. You should be more concerned that the generated configs do not pass YAML lint than that the test files do not pass.

At today's stage, no this is not what I am concerned about. Via this commit, I simply want to ensure the yaml files on this repo are valid yaml (independently of how AWX manipulate yaml) - and enforce only valid yaml files (tests files or not) are submitted to the repo by enabling a new lint check.

The issue this test update is solving is that the test was comparing string `---a:b\n` vs. `a:b\n` rather than actual yaml objects (when compared as string the precedent comparison yields to false, while those two yaml objects are indeed the same)

Reply:
The point of those files is to get everyone on the same page about what content is being generated. If we merge this, the content of those files will not be the content that is being generated. Could we ignore them from the check instead?

Reply:
I tend to agree with @AlanCoding, I don't think it's important to verify the linting of the YAML generated by inventory sources.